### PR TITLE
CO-2323_CoreAPI_create

### PR DESCRIPTION
**Pull request by ioigoume**
CoreAPI Create User

### CODE REVIEW

File: app/Model/AppModel.php [hunk 1]
```diff
@@ -1195,7 +1195,76 @@ public function isPlugin($type = null) {
...
+   * Returns false if any fields passed match any (by default, all if $or = false) of their matching values.
+   *
+   * Can be used as a validation method. When used as a validation method, the `$or` parameter
+   * contains an array of fields to be validated.
+   *
+   * @param array $fields Field/value pairs to search (if no values specified, they are pulled from $this->data)
+   * @param bool|array $or If false, all fields specified must match in order for a false return value
+   * @return bool False if any records matching any fields are found
+   */
+  public function isUniqueChangelog($fields, $or = true) {
```
Comment (boshrin) [inline app/Model/AppModel.php:1212]: I'm not sure why we're adding this check? We support multiple copies of the same EmailAddress or Identifier on the same record, for example this is a common use case when syncing records from multiple Org Identity Sources.

File: app/Plugin/CoreApi/Config/routes.php [hunk 2]
```diff
@@ -71,16 +71,10 @@
...
-// Scoped identifiers are not parsed properly because they are perceived as file extensions
-// Enable extensions parse to resolve this problem
-Router::parseExtensions('json', 'xml');
 
-
-/*
-
-Router::connect(
 // XXX This needs to trigger identifier assignment and maybe some other stuff
 //     provisioning should only fire after all models are saved
```
Comment (boshrin) [inline app/Plugin/CoreApi/Config/routes.php:76]: It looks like you did this, so we can probably delete this comment?

File: app/Plugin/CoreApi/Model/CoreApi.php [hunk 3]
```diff
@@ -180,6 +180,214 @@ public function cmPluginMenus() {
...
+      // Assign Identifiers
+      // Disable provisioning since we will bulk provision at the end
+      // This will return an array describing which, if any, identifiers were assigned,
+      // but we don't do anything with the result here
+      $this->Co->CoPerson->Identifier->assign('CoPerson', $co_person_id, null, false, $actorApiUserId);
+
+      // Handle plugin models
+      // - Authenticator
+      // - XXX Clusters currently not supported
+      $not_supported_plugins = array('UnixCluster');
```
Comment (boshrin) [inline app/Plugin/CoreApi/Model/CoreApi.php:342]: Why aren't clusters supported? And why is only the UnixCluster plugin excluded?

File: app/Plugin/CoreApi/Controller/ApiController.php [hunk 4]
```diff
@@ -211,6 +212,31 @@ public function beforeFilter() {
...
 
+  /**
+   * Handle a Core API CO Person record create
+   *
+   * @since  COmanage Registry v4.1.0
+   */
+
+  public function create() {
+    try {
+      $ret = $this->CoreApi->createV1($this->cur_api['CoreApi']['co_id'],
```
Comment (boshrin) [inline app/Plugin/CoreApi/Controller/ApiController.php:223]: `createV1` doesn't appear to actually return anything? We need to convey the Identifier of the configured Identifier Type in the response.

File: app/Plugin/CoreApi/Config/routes.php (hunk 2 above)
Comment (ioigoume) [reply app/Plugin/CoreApi/Config/routes.php]: Done

File: app/Plugin/CoreApi/Controller/ApiController.php (hunk 4 above)
Comment (ioigoume) [reply app/Plugin/CoreApi/Controller/ApiController.php:223]: done

File: app/Plugin/CoreApi/Model/CoreApi.php (hunk 3 above)
Comment (ioigoume) [reply app/Plugin/CoreApi/Model/CoreApi.php]: I am not sure why i added this check there. I removed  it.

File: app/Model/AppModel.php (hunk 1 above)
Comment (ioigoume) [reply app/Model/AppModel.php:1212]: @boshrin let's discuss this over the Dev call.

Comment (ioigoume) [reply app/Model/AppModel.php:1212]: @boshrin Removed the `isUniqueChangelog` validation rule from the `EmailAddress` and the `Url` Model

**Pull request merged**